test: WaterfallTestClass completion E2E test flaky — falls back to word-based completions

## Description

The `Waterfall Loading E2E Tests > Completion shows class definitions` test is flaky in the `reliability-workflows` E2E category. It fails intermittently because the completion provider returns word-based fallback completions instead of symbol-based completions that include `WaterfallTestClass`.

## Problem

The `Waterfall Loading E2E Tests > Completion shows class definitions` test is flaky. On some runs it fails because the completion provider returns alphabetical word-based completions (`A, actually, After, after, and, Another...`) instead of symbol-based completions that include `WaterfallTestClass`.

**Test location:** `packages/vscode-pike/src/test/integration/lsp-features.test.ts:1851-1875`

**Fixture:** `packages/vscode-pike/test-workspace/test-waterfall-completion.pike`

**Error:**
```
AssertionError: Completion should show "WaterfallTestClass". Got: A, actually, After, after, and, Another, appear, are, available, basic, be, body, both, Class, class, class_field, class_method, Completion, completion, constant
```

## Expected Behavior

The completion provider returns `WaterfallTestClass` in the completion list when the class is defined in the current file.

## Suggested Approach

Replace the single-shot completion assertion with a `waitFor` retry loop. The test should retry the completion request until `WaterfallTestClass` appears in the results, accommodating the indexing race without changing server behavior.

## Affected Files

- packages/vscode-pike/src/test/integration/lsp-features.test.ts
- packages/vscode-pike/test-workspace/test-waterfall-completion.pike

## Acceptance

- [x] Flaky test passes reliably on CI
- [ ] Root cause documented in code comments

--- a/packages/vscode-pike/src/test/integration/lsp-features.test.ts
+++ b/packages/vscode-pike/src/test/integration/lsp-features.test.ts
@@ -1847,31 +1847,36 @@ suite('Waterfall Loading E2E Tests', () => {
 
   /**
    * Test: Completion shows class definitions from current file
+   *
+   * This test retries because the completion provider may not have indexed
+   * the file yet when the first completion request arrives. The suiteSetup
+   * waits for document symbols, but the completion pipeline has its own
+   * indexing path that can lag behind. Retry until WaterfallTestClass appears
+   * or timeout (15s).
    */
   test('Completion shows class definitions', async function () {
     this.timeout(30000);
 
     const position = positionForRegex(document, /WaterfallTestClass obj =/m);
 
-    const completions = await vscode.commands.executeCommand<vscode.CompletionList>(
-      'vscode.executeCompletionItemProvider',
-      testDocumentUri,
-      position
+    // Retry until the completion provider has indexed WaterfallTestClass.
+    // The provider may return word-based completions on first request while
+    // symbol indexing is still in progress.
+    const result = await waitFor(
+      'WaterfallTestClass in completions',
+      () => vscode.commands.executeCommand<vscode.CompletionList>(
+        'vscode.executeCompletionItemProvider',
+        testDocumentUri,
+        position
+      ),
+      (completions) => {
+        if (!completions || completions.items.length === 0) return false;
+        return completions.items.map(labelOf).includes('WaterfallTestClass');
+      },
+      15000
     );
 
-    assert.ok(completions, 'Should return completions');
-    assert.ok(completions!.items.length > 0, 'Should have completion items');
-
-    const completionLabels = completions!.items.map(labelOf);
-
-    // WaterfallTestClass should appear
-    const hasClass = completionLabels.includes('WaterfallTestClass');
-
-    assertWithLogs(
-      hasClass,
-      'Completion should show "WaterfallTestClass". Got: ' +
-        completionLabels.slice(0, 20).join(', ')
-    );
+    assert.ok(result, 'Should return completions');
   });
 
   /**

